service/src/main/java/greencity/service/NotificationServiceImpl.java

```diff
@@ -98,5 +119,12 @@ public void deleteNotification(Long notificationId) {
         notificationRepo.deleteById(notificationId);
     }
 
+    @Override
+    public int countUnreadNotifications(Long userId) {
+        User receiver = findUserById(userId);
+        return notificationRepo.findUnreadByReceiver(receiver).size();
```

Comment:
this could be a method in the repo so that the sql query is `select count(*) from notification where ...` instead of `select * from notification where ...`